### PR TITLE
Configurable cooldown for /email recovery

### PR DESCRIPTION
**Pull request by ljacqu**
Impl for #1073

- Add configurable cooldown period after sending an email for /email recovery
- Create method for translatable duration information (e.g. "20 minutes")

### CODE REVIEW

**Comment by ljacqu** [comment]
hi @Gnat008, can I bug you to have a look at this if you have some time for me? :)

File: src/main/java/fr/xephi/authme/command/executable/email/RecoverEmailCommand.java [hunk 1]
```diff
@@ -78,15 +96,29 @@ public void runCommand(Player player, List<String> arguments) {
                 processRecoveryCode(player, arguments.get(1), email);
             }
         } else {
-            generateAndSendNewPassword(player, email);
+            boolean maySendMail = checkEmailCooldown(player);
```
Comment (Gnat008) [inline src/main/java/fr/xephi/authme/command/executable/email/RecoverEmailCommand.java]: Here you check the cooldown, but in the `createAndSendRecoveryCode(Player, String)` method you do the check inside the method, and not before calling it. Would it make sense to put this check in the `generateAndSendNewPassword(Player, String)` instead?

Comment (ljacqu) [inline src/main/java/fr/xephi/authme/command/executable/email/RecoverEmailCommand.java]: @Gnat008 Thanks for the review—good catch! The reason this is not checked in this path is that the user can only trigger an email after having entered the correct recovery code. Moreover, if we configure a timeout of 5 minutes and the user is faster than that in reading his mail we would force him to wait ~3 minutes before he can enter the code. 
This is pretty much spaghetti code at this point and I really dislike that an ExecutableCommand is keeping state (and implementing `Reloadable`!) With #1109 we should improve these things.

Comment (Gnat008) [inline src/main/java/fr/xephi/authme/command/executable/email/RecoverEmailCommand.java]: Sounds good to me, thanks for the explanation!

**Comment by ljacqu** [comment]
Feels weird to merge this after I just admitted that this is spaghetti code... :smile:

**Pull request merged**